src/main/java/com/powsybl/openloadflow/sensi/DcSensitivityAnalysis.java

```diff
@@ -539,6 +540,43 @@ private void processConverterStation(LfNetwork lfNetwork, HvdcConverterStation<?
 
     }
 
+    private Map<Generator, GeneratorState> applyGeneratorContingency(Network network, PropagatedContingency contingency) {
+        // It applies on the network the loss of the generators contained in the contingency.
+        // Generator states are stored.
+        Collection<Generator> generatorsToSave = new HashSet<>();
+        for (String generatorId : contingency.getGeneratorIdsToLose()) {
+            Generator generator = network.getGenerator(generatorId);
+            generatorsToSave.add(generator);
+        }
+        Map<Generator, GeneratorState> generatorStates = GeneratorState.createGeneratorStates(generatorsToSave);
+        for (String generatorId : contingency.getGeneratorIdsToLose()) {
```

Comment:
You have to do that on the `LfGenerator` object directly and in the same time you have to remove it from slack distribution if in the parameters slack is distributed and the distribution is made on generators.